test/gator/test/test.bats

```diff
@@ -260,3 +260,26 @@ match_yaml_msg () {
   match_substring "${output[*]}" "${want_msg_1}"
   match_substring "${output[*]}" "${want_msg_2}"
 }
+
+@test "expansion with namespace selector" {
+  # First run without the namespace and expect an err
+  run bin/gator test \
+    -f="$BATS_TEST_DIRNAME/fixtures/manifests/expansion/expansion-w-ns-selector.yaml" \
+    -o=yaml
+
+  [ "$status" -eq 1 ]
+
+  want_msg="[Implied by expand-deployments] unable to match constraints: error matching the requested object: failed to run Match criteria: namespace selector for namespace-scoped object but missing Namespace"
```

Comment:
How can the user tell which resource and/or constraint is the unhappy one?

Reply:
good point. For the mutation case, I added a neat error message that points to which mutator failed the match and for which obj.

In the constraint case, it may be a bit tricky to get the same richness for now.

Let's break the err message down:

`unable to match constraints: error matching the requested object: failed to run Match criteria: namespace selector for namespace-scoped object but missing Namespace`

* `unable to match constraints` comes from the framework; printing out which constraint failed will need a framework change which I am happy to make but cannot do so in this PR since it's a different repo.

* `error matching the requested object` comes from the `target/matcher.go` pkg in g8r. I could amend these lines to include which obj failed.

The best way to improve the UX imo, would be to make the two changes above. I can make the latter in this PR and the former in a PR to CF.

Reply:
Can we print out the name of the resource that is raising the constraint matcher error without modifying CF?

Reply:
Yes, as of https://github.com/open-policy-agent/gatekeeper/pull/2812/commits/698a5cf99f123487929d9e1b78dcbf5cd2c65103 I added the resource name to be printed out of the matcher in gatekeeper.

Example from bats test: see `nginx-deployment-pod` below:

```
Implied by expand-deployments] unable to match constraints: error matching the requested object: nginx-deployment-pod :failed to run Match criteria: namespace selector for namespace-scoped object but missing Namespace
```

Reply:
This `error matching the requested object: nginx-deployment-pod` line would appear for a constraint failing to match too, right?

Reply:
Yes that example is for a constraint failing the match in particular.